### PR TITLE
Rails Blog: Validation spec

### PR DESCRIPTION
**Pull request by Profsain**
In this part of Rails blog project, I implemented the following requirement
- Created model validation
- Created model validation spec

### CODE REVIEW

**Comment by TedLivist** [comment]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @Profsain,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project with a failing linter. Kindly fix your rubocop

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    it 'should return user last 3 posts' do
+      User.create(
+        name: 'John Doe',
+        photo: 'https://picsum.photos/200/300',
+        bio: 'Teacher from Nigeria',
+        posts_counter: 0
+      )
+    end
+  end
```
Comment (TedLivist) [inline spec/models/user_spec.rb]: - [ ] Kindly write your test for the `#fetch_recent_posts` method properly. There should be an assertion (like `expect`) that verifies that the last three posts are actually fetched.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  it '@post.likes_counter is an integer' do
+    expect(@post.likes_counter).to be_an(Integer)
+  end
+
+  it '@post.likes_counter is greater than or equal to 0' do
+    expect(@post.likes_counter).to be >= 0
+  end
+end
```
Comment (TedLivist) [inline spec/models/post_spec.rb:36]: - [ ] All custom methods in the model should be tested. Kindly add a test for the `#fetch_recent_comments` method in the Post model. Ensure there is an assertion verifying that the last 5 comments belonging to the post are fetched

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Profsain,

Good job so far!
✔️ Addition of validations in the model
✔️ Passing tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello  @Profsain        , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Descriptive P.R







 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

**Pull request merged**